### ISSUE TITLE
Death on launch

### ISSUE THREAD

**Issue by josmithiii** [spyder-ide/spyder#14800]
### What steps will reproduce the problem?

Launch by typing "spyder" in a Terminal window.
Originally installed by MacPorts, but this version downloaded 
and installed from .dmg file.

### Traceback
```python-traceback
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/spyder/plugins/editor/widgets/status.py", line 113, in process_git_data
    branches, branch, files_modified = output
TypeError: cannot unpack non-iterable NoneType object
```

## Versions

* Spyder version: 4.2.1 
* Python version: 3.8.8
* Qt version: 5.15.2
* PyQt5 version: 5.15.2
* Operating System: Darwin 20.3.0

### Dependencies

```

# Mandatory:
applaunchservices >=0.1.7       :  0.2.1 (OK)
atomicwrites >=1.2.0            :  1.4.0 (OK)
chardet >=2.0.0                 :  4.0.0 (OK)
cloudpickle >=0.5.0             :  1.6.0 (OK)
diff_match_patch >=20181111     :  20200713 (OK)
intervaltree >=3.0.2            :  3.1.0 (OK)
IPython >=7.6.0                 :  7.20.0 (OK)
jedi >=0.17.2                   :  0.18.0 (OK)
jsonschema >=3.2.0              :  3.2.0 (OK)
keyring >=17.0.0                :  22.0.1 (OK)
nbconvert >=4.0                 :  6.0.7 (OK)
numpydoc >=0.6.0                :  1.1.0 (OK)
parso >=0.7.0                   :  0.8.1 (OK)
pexpect >=4.4.0                 :  4.8.0 (OK)
pickleshare >=0.4               :  0.7.5 (OK)
psutil >=5.3                    :  5.8.0 (OK)
pygments >=2.0                  :  2.8.0 (OK)
pylint >=1.0                    :  2.6.0 (OK)
pyls >=0.36.2;<1.0.0            :  0.36.2 (OK)
pyls_black >=0.4.6              :  0.4.6 (OK)
pyls_spyder >=0.3.0             :  0.3.2 (OK)
qdarkstyle >=2.8                :  2.8.1 (OK)
qtawesome >=0.5.7               :  1.0.2 (OK)
qtconsole >=5.0.1               :  5.0.2 (OK)
qtpy >=1.5.0                    :  1.9.0 (OK)
setuptools >=39.0.0             :  51.3.3 (OK)
sphinx >=0.6.6                  :  3.5.1 (OK)
spyder_kernels >=1.10.1;<1.11.0 :  1.10.1 (OK)
textdistance >=4.2.0            :  4.2.1 (OK)
three_merge >=0.1.1             :  0.1.1 (OK)
watchdog >=0.10.3               :  1.0.2 (OK)
zmq >=17                        :  22.0.3 (OK)

# Optional:
cython >=0.21                   :  0.29.21 (OK)
matplotlib >=2.0.0              :  3.3.4 (OK)
numpy >=1.7                     :  1.20.1 (OK)
pandas >=1.1.1                  :  1.2.2 (OK)
scipy >=0.17.0                  :  1.6.0 (OK)
sympy >=0.7.3                   :  1.7.1 (OK)
```

**Comment by josmithiii** [comment]
I see now that I launched the older MacPorts version.  When I launch the newly installed app in /Applications/, there is no issue on launch.  I guess I should uninstall the macports version now.

**Comment by ccordoba12** [comment]
Hey @josmithiii, thanks for reporting. This problem was fixed in our 4.2.2 version, released today. Once that version is part of MacPorts, you won't see this problem again.
